src/topp/MapAlignerTreeGuided.cpp

```diff
@@ -0,0 +1,581 @@
+    model_params = model_params.copy(model_type+":", true);
+
+    //-------------------------------------------------------------
+    // reading input
+    //-------------------------------------------------------------
+    Size in_files_size = in_files.size();
+    FeatureXMLFile fxml_file;
+    // define here because needed to load and store
+    FeatureFileOptions param = fxml_file.getOptions();
+    // to save memory don't load convex hulls and subordinates
+    param.setLoadSubordinates(false);
+    param.setLoadConvexHull(false);
+    fxml_file.setOptions(param);
+
+    vector<FeatureMap> feature_maps(in_files_size);
+    loadInputMaps_(feature_maps, in_files, fxml_file);
+
+    //-------------------------------------------------------------
+    // calculations
+    //-------------------------------------------------------------
+
+    vector<vector<double>> maps_ranges(in_files_size);  // to save ranges for alignment (larger rt_range -> reference)
+    std::vector<BinaryTreeNode> tree;    // to construct tree with pearson coefficient
+    buildTree_(feature_maps, tree, maps_ranges);
+
+    // print tree
+    ClusterAnalyzer ca;
+    OPENMS_LOG_INFO << "  Alignment follows tree: " << ca.newickTree(tree, true) << endl;
+
+    vector<Size> trafo_order;
+    FeatureMap map_transformed;
+    // alignment uses copy of feature_maps, returning result within map_transformed (to keep original data)
+    treeGuidedAlignment_(tree, feature_maps, maps_ranges, map_transformed, trafo_order, model_params, model_type);
+
+    //-------------------------------------------------------------
+    // generating output
+    //-------------------------------------------------------------
+    vector<TransformationDescription> transformations(in_files_size); // for trafo_out
+    computeTransformationsByOrigInFeature_(feature_maps, map_transformed, transformations, trafo_order, model_params, model_type);
+    computeTransformedFeatureMaps_(feature_maps, transformations);
+
+    //-------------------------------------------------------------
+    // writing output
+    //-------------------------------------------------------------
+    // store transformed feature_maps
+    storeFeatureXMLs_(feature_maps, out_files, fxml_file);
+
+    // store transformations
+    storeTransformationDescriptions_(transformations, out_trafos);
+
+    return EXECUTION_OK;
+  }
+};
+
+template<typename MapType>
+void TOPPMapAlignerTreeGuided::loadInputMaps_(vector<MapType>& maps, StringList& ins, FeatureXMLFile& fxml_file)
+{
+  ProgressLogger progresslogger;
+  progresslogger.setLogType(TOPPMapAlignerBase::log_type_);
+  progresslogger.startProgress(0, ins.size(), "loading input files");
+  for (Size i = 0; i < ins.size(); ++i)
+  {
+    progresslogger.setProgress(i);
+    fxml_file.load(ins[i], maps[i]);
+  }
+  progresslogger.endProgress();
+}
+
+void TOPPMapAlignerTreeGuided::getPeptideSequences_(const vector<PeptideIdentification>& peptides,
```

Comment:
since the output vectors are not cleared, 'addPeptideSequences_' would be a better name?